aptos-move/framework/aptos-framework/sources/Account.move

```diff
@@ -343,18 +343,70 @@ module AptosFramework::Account {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    /// Basic account creation method.
+    /// Basic account creation methods.
     ///////////////////////////////////////////////////////////////////////////
 
     public(script) fun create_account(auth_key: address) {
-        let (signer, _) = create_account_internal(auth_key);
+        let signer = create_account_internal(auth_key);
         Coin::register<TestCoin>(&signer);
     }
 
+    public fun create_module_account(
+        source: &signer,
+    ): (signer, SignerCapability) {
+        let guid = GUID::create(source);
+        let bytes = BCS::to_bytes(&guid);
+        Vector::append(&mut bytes, BCS::to_bytes(&Timestamp::now_microseconds()));
+        let addr = create_address(Hash::sha3_256(bytes));
```

Comment:
Nice! This is the same way contract addresses are created in EVM - hashes of the deployer accounts. Can we potentially use the account's nonce here if that's available? Does GUID::create(source) guarantee uniqueness here among multiple modules belonging to the same account?

Reply:
we could eliminate the timestamp in microseconds if we want that. I think we don't need that as I can imagine the flow would be:

```
(signer, cap) = create_module_account()
dao = Dao { signer_cap: cap, .... };
move_to(&signer, dao)
```

Reply:
You didn't really answer any of my questions :) but upon looking at GUID, it also uses some concept of counter already.